examples/stream/Cargo.toml

```diff
@@ -0,0 +1,23 @@
+[package]
+name = "stream-example"
+version = "0.1.0"
+edition = "2021"
+publish = false
+license = "MIT"
+
+[package.metadata.release]
+release = false
+
+[dependencies]
+anyhow = "1"
+futures = "0.3.29"
+libp2p = { path = "../../libp2p", features = [ "tokio", "quic"] }
+tokio = { version = "1.35", features = ["full"] }
+tracing = "0.1.37"
+tracing-subscriber = { version = "0.3", features = ["env-filter"] }
+rand = "0.8"
+flume = "0.11.0"
```

Comment:
If I understand correctly, we only need `flume` to do work stealing. Is that correct?

I would argue that the benefit of work-stealing does not warrant the additional dependency. Instead I would suggest a `futures::channel::mpsc` to each `ConnectionHandler`. The channels each can have a capacity of `0` (effectively `1` given that each sender gets a slot). the `NetworkBehaviour` iterates through all channels and sees if one has capacity.

Reply:
Going to resolve this in favor of the discussion above. It is not just about work-stealing although I do find that it plays well together. I am open to removing `flume` again if we can solve the above in a clean way. `flume` is also a very small dependency so I don't think it matters much.